pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -135,33 +144,35 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
 	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers *cache.SocketPluginHandlers,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
-		client, conn, err := dial(socketPath, dialTimeoutDuration)
+		_, conn, err := dial(socketPath, dialTimeoutDuration)
 		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+			klog.V(4).Infof("unable to dial: %v", err)
+		} else {
+			conn.Close()
 		}
-		defer conn.Close()
 
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-		defer cancel()
+		var handlerWithName cache.NamedPluginHandler
+		pathToHandlers.Lock()
+		handlerWithName, handlerFound := pathToHandlers.Handlers[socketPath]
+		pathToHandlers.Unlock()
 
-		infoResp, err := client.GetInfo(ctx, &registerapi.InfoRequest{})
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- failed to get plugin info using RPC GetInfo at socket %s, err: %v", socketPath, err)
+		if !handlerFound {
+			return fmt.Errorf("unable to DeRegister - there is no handler for %s", socketPath)
```

Comment:
Just one last nit so the error is consistent with the other functions:
```suggestion
			return fmt.Errorf("UnregisterPlugin error -- failed to get plugin handler for %s", socketPath)
```

Reply:
Addressed.
Please take another look, @taragu .